File: leetcode/src/main/golang/longest_common_prefix.go
import "strings"

func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }
    tmp := strs[0]
    for i := 1; i < len(strs); i++ {
        for ; strings.Index(strs[i], tmp) != 0; {
            tmp = tmp[0:len(tmp) - 1]
        }
    }
    return tmp
}

File: leetcode/src/main/ruby/longest_common_prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
    return "" if strs.length == 0
    i, tmp = 1, strs[0]
    while i < strs.length
        tmp = tmp[0...-1] while strs[i].index(tmp) != 0
        i += 1
    end
    tmp
end